### PR TITLE
Company Info Helper

### PR DESCRIPTION
**Pull request by smilesthesad**
## Summary of Changes

Adds a helper that, given a crunchbase entity id or a crunchbase company page, returns info on the company. Currently returns the following information:
```
{
  properties: {
    short_description: 'Airbnb is an online community marketplace for people to list, discover, and book accommodations through mobile phones or the internet.',
    identifier: {
      uuid: 'bcb617c3-9e43-d5b0-1d14-82b795f2642f',
      value: 'Airbnb',
      image_id: 'bomkrvppqn1pssoqsxqy',
      permalink: 'airbnb',
      entity_def_id: 'organization'
    }
  }
}
```
## Motivation and Explanation

Works towards <https://github.com/uwcsc/codeybot/issues/122>. Once this is in, we can track companies in our database and start implementing commands.
## Related Issues

Resolves <https://github.com/uwcsc/codeybot/issues/402>. 
## Steps to Reproduce

Call the function through ts-node. 
You might not be able to import modules, so you can comment out the imports and replace them with 
```
const fetch = require('node-fetch');
const { isValidUrl } = require('./validateUrl');
const { vars } = require('../config');
``` 
for testing
## Demonstration of Changes

https://user-images.githubusercontent.com/36215359/203208004-0c59cbfa-0f5e-4dd6-894d-d42e004122e6.mov

### CODE REVIEW

File: src/utils/companyinfo.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  const response = await fetch(
+    `https://api.crunchbase.com/api/v4/entities/organizations/${companyName}?user_key=${vars.CRUNCHBASE_API_KEY}&field_ids=categories,short_description`,
+  );
+  const data = await response.json();
+  if (data.error) {
+    throw new Error(data.error);
+  }
+  console.log(data);
```
Comment (Picowchew) [inline src/utils/companyinfo.ts:33]: Just for the sake of making a commit, I believe we can remove this `console.log` statement.

**Review by Picowchew — CHANGES_REQUESTED** [review]
Overall looks good! I just left some very small nitpicks.

**Review by Picowchew — CHANGES_REQUESTED** [review]
Oh one more thing, the file `companyinfo.ts` should be renamed to `companyInfo.ts`.

File: src/utils/companyinfo.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { vars } from '../config';
+import { isValidUrl } from './validateUrl';
+import fetch from 'node-fetch';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const getCompanyInfo = async (companyCrunchbaseLink: string) => {
+  // check if url contains crunchbase.com/organization
+  // if not, return null
```
[the remark this answers is not in the document]
Comment (smilesthesad) [reply src/utils/companyinfo.ts:8]: good catch

File: src/utils/companyinfo.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  // if not, throw error
+  if (
+    isValidUrl(companyCrunchbaseLink) &&
+    !companyCrunchbaseLink.includes(CRUNCHBASE_ORGANIZATION_URL)
+  ) {
+    throw new Error('Invalid URL');
+  }
+
+  let companyName;
+  // check if urlPattern matches companyCrunchbaseLink
```
Comment (Picowchew) [inline src/utils/companyinfo.ts:20]: Isn't this a misplaced comment, since `urlPattern` is found in `isValidUrl`?

**Review by Picowchew — CHANGES_REQUESTED** [review]

**Review by Picowchew — APPROVED** [review]
LGTM!

**Review by mcpenguin — APPROVED** [review]
LGTM too!

**Pull request merged**